flink-table/flink-table-common/src/test/java/org/apache/flink/table/catalog/GenericInMemoryCatalogTest.java

```diff
@@ -0,0 +1,581 @@
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test for GenericInMemoryCatalog.
+ */
+public class GenericInMemoryCatalogTest {
+	private static final String IS_STREAMING = "is_streaming";
+
+	private final String testCatalogName = "test-catalog";
+	private final String db1 = "db1";
+	private final String db2 = "db2";
+
+	private final String t1 = "t1";
+	private final String t2 = "t2";
+	private final ObjectPath path1 = new ObjectPath(db1, t1);
+	private final ObjectPath path2 = new ObjectPath(db2, t2);
+	private final ObjectPath path3 = new ObjectPath(db1, t2);
+	private final ObjectPath nonExistDbPath = ObjectPath.fromString("non.exist");
+	private final ObjectPath nonExistObjectPath = ObjectPath.fromString("db1.nonexist");
+
+	private static final String TEST_COMMENT = "test comment";
+
+	private static ReadableWritableCatalog catalog;
+
+	@Before
+	public void setUp() {
+		catalog = new GenericInMemoryCatalog(db1);
+	}
+
+	@Rule
+	public ExpectedException exception = ExpectedException.none();
+
+	@After
+	public void close() {
+		if (catalog.tableExists(path1)) {
+			catalog.dropTable(path1, true);
+		}
+		if (catalog.tableExists(path2)) {
+			catalog.dropTable(path2, true);
+		}
+		if (catalog.tableExists(path3)) {
+			catalog.dropTable(path3, true);
+		}
+		if (catalog.databaseExists(db1)) {
+			catalog.dropDatabase(db1, true);
+		}
+		if (catalog.databaseExists(db2)) {
+			catalog.dropDatabase(db2, true);
+		}
+	}
+
+	@AfterClass
+	public static void clean() throws IOException {
+		catalog.close();
+	}
+
+	// ------ tables ------
+
+	@Test
+	public void testCreateTable_Streaming() {
+		catalog.createDatabase(db1, createDb(), false);
+		GenericCatalogTable table = createStreamingTable();
+		catalog.createTable(path1, table, false);
+
+		CatalogTestUtil.compare(table, (GenericCatalogTable) catalog.getTable(path1));
+	}
+
+	@Test
+	public void testCreateTable_Batch() {
+		catalog.createDatabase(db1, createDb(), false);
+
+		GenericCatalogTable table = createTable();
+		catalog.createTable(path1, table, false);
+
+		CatalogTestUtil.compare(table, (GenericCatalogTable) catalog.getTable(path1));
+
+		List<String> tables = catalog.listTables(db1);
+
+		assertEquals(1, tables.size());
+		assertEquals(path1.getObjectName(), tables.get(0));
+
+		catalog.dropTable(path1, false);
+	}
+
+	@Test
+	public void testCreateTable_DatabaseNotExistException() {
+		assertFalse(catalog.databaseExists(db1));
+
+		exception.expect(DatabaseNotExistException.class);
```

Comment:
Add exception.expectMessage() to verify error message. Same for the tests below.

Reply:
Already added.